pkg/operation/care/checker.go

```diff
@@ -258,13 +261,26 @@ func resourcesNotProgressingCheck() func(status gardencorev1beta1.ConditionStatu
 	}
 }
 
-func (b *HealthChecker) checkManagedResourceConditions(condition gardencorev1beta1.Condition, mr *resourcesv1alpha1.ManagedResource, conditionsToCheck map[gardencorev1beta1.ConditionType]func(status gardencorev1beta1.ConditionStatus) bool) *gardencorev1beta1.Condition {
+func (b *HealthChecker) checkManagedResourceConditions(
+	condition gardencorev1beta1.Condition,
+	mr *resourcesv1alpha1.ManagedResource,
+	conditionsToCheck map[gardencorev1beta1.ConditionType]func(status gardencorev1beta1.ConditionStatus) bool) *gardencorev1beta1.Condition {
 	if mr.Generation != mr.Status.ObservedGeneration {
 		c := b.FailedCondition(condition, gardencorev1beta1.OutdatedStatusError, fmt.Sprintf("observed generation of managed resource '%s/%s' outdated (%d/%d)", mr.Namespace, mr.Name, mr.Status.ObservedGeneration, mr.Generation))
 		return &c
 	}
 
+	var conditionProgressing, conditionHealthy bool
 	for _, cond := range mr.Status.Conditions {
+		if cond.Type == resourcesv1alpha1.ResourcesProgressing &&
+			cond.Status == gardencorev1beta1.ConditionTrue &&
+			b.managedResourceProgressingThreshold != nil &&
+			time.Since(cond.LastTransitionTime.Time) >= b.managedResourceProgressingThreshold.Duration {
+			conditionProgressing = true
```

Comment:
Why do we set this variable to `true`? This is confusing - shouldn't we use a new variable for this scenario?